centrallix/osdrivers/objdrv_report_v3.c

```diff
@@ -4143,6 +4240,509 @@ rpt_internal_DoChart(pRptData inf, pStructInf chart, pRptSession rs, int contain
 	    nmFree(ctx, sizeof(RptChartContext));
 	return errval;
     }
+/*** rpt_internal_DoAutoChart prints a bar, line, or pie chart on the report based on the
+ *** information provided by the user. Series are automatically broken up by the series_value pulled from the query.
+ *** MathGL is used for the rendering.
+ ***/
+int
+rpt_internal_DoAutoChart(pRptData inf, pStructInf chart, pRptSession rs, int container_handle)
+    {
+    const char* AUTO_CHART_NAME = "report/chart-auto-series";
+    int errval = -1; /* default */
+    pRptActiveQueries ac = NULL;
+    int rval;
+    int i,j;
+    pPrtImage img = NULL;;
+    int flags;
+    double x,y,w,h;
+    char* chart_type;
+    char* title;
+    char* x_axis_label;
+    char* y_axis_label;
+    char* ptr;
+    double* dbl_ptr = NULL;
+    int* int_ptr = NULL;
+    int series_index;
+    char* cur_x = NULL;
+    pXArray tempVals = NULL;
+    pXArray tempSeries = NULL;
+    int box = 0;
+    int stacked;
+    int axis_fontsize;
+    pStructInf subobj, childobj;
+    pExpression childexp;
+    pRptChartValues value = NULL;
+    pRptChartDriver drv;
+    pRptChartContext ctx = NULL;
+    double xoffset, yoffset;
+    char color[8];
+    char* paletteBuf = NULL;
+    char* start;
+    char* end;
+    int series_cnt, paletteInd;
+    pXArray palette = NULL;
+    int prec;
+    char precstr[32];
+    
+    #ifndef HAVE_MGL2
+	// Only supporting mathgl2 
+	mssError(0, "RPT", "auto-chart requires MGL2");
+	goto error;
+    #endif
+
+	/** Conditional rendering of the chart **/
+	rval = rpt_internal_CheckCondition(inf, chart);
+	if (rval < 0) return rval;
+	if (rval == 0) return 0;
+
+	/** Determine chart type **/
+	if (rpt_internal_GetString(inf, chart, "chart_type", &chart_type, NULL, 0) < 0)
+	    {
+	    mssError(0, "RPT", "Chart type required for chart '%s'", chart->Name);
+	    goto error;
+            }
+	for(i=0; i<RPT_INF.ChartDrivers.nItems; i++)
+	    {
+	    drv = (pRptChartDriver)RPT_INF.ChartDrivers.Items[i];
+	    if (!strcmp(drv->Type, chart_type))
+		break;
+	    drv = NULL;
+	    }
+	if (!drv)
+	    {
+	    mssError(0, "RPT", "Invalid chart type '%s' for chart '%s'", chart_type, chart->Name);
+	    goto error;
+	    }
+
+	/** Set up rendering context **/
+	ctx = (pRptChartContext)nmMalloc(sizeof(RptChartContext));
+	if (!ctx)
+	    goto error;
+	memset(ctx, 0, sizeof(RptChartContext));
+	ctx->inf = inf;
+	rpt_internal_GetString(inf, chart, "show_legend", &ptr, "no", 0);
```

Comment:
This can use rpt_internal_GetBool().

Reply:
That makes the code simpler - thank you!